controllers/message.go

```diff
@@ -153,12 +153,20 @@ func (c *ApiController) AddMessage() {
 		c.ResponseError(err.Error())
 		return
 	}
-	if message.IsRegenerated {
-		messages, err := object.GetChatMessages(message.Chat)
-		if err != nil {
-			c.ResponseError(err.Error())
-			return
-		}
+
+	addMessageAfterSuccess := true
+	messages, err := object.GetChatMessages(message.Chat)
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	if message.IsRegenerated && len(messages) == 1 && messages[0].ReplyTo == "Welcome" && messages[0].Author == "AI" {
+		object.DeleteMessage(messages[0])
+		message.Author = "AI"
+		message.ReplyTo = "Welcome"
+		addMessageAfterSuccess = false
```

Comment:
if doesn't need, just return

Reply:
After this it's still necessary to run object.AddMessage() one more time.